### PR TITLE
FISH-6285 Apply patches

### PR DESCRIPTION
**Pull request by kalinchan**

### CODE REVIEW

File: ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/InterfaceModel.java [hunk 1]
```diff
@@ -24,13 +24,12 @@
 import java.util.List;
```
Comment (luiseufrasio) [inline ext/microprofile/mp-rest-client/src/main/java/org/glassfish/jersey/microprofile/restclient/InterfaceModel.java:24]: update Copyright year

File: tests/osgi/functional/src/test/java/org/glassfish/jersey/osgi/test/basic/ApacheOsgiIntegrationTest.java [hunk 2]
```diff
@@ -45,7 +45,7 @@
 /**
  * @author Adam Lindenthal
  */
-@RunWith(PaxExam.class)
+//@RunWith(PaxExam.class)
```
Comment (luiseufrasio) [inline tests/osgi/functional/src/test/java/org/glassfish/jersey/osgi/test/basic/ApacheOsgiIntegrationTest.java]: This test will not be executed? Why?

**Review by luiseufrasio — DISMISSED** [review]

**Review by Pandrex247 — COMMENTED** [review]
MicroProfile Rest Client TCK doesn't seem to be happy with this. Lots of NoClassDefFounds, can you double-check?
`java.lang.NoClassDefFoundError: org/glassfish/jersey/ext/cdi1x/internal/CdiUtil`

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
Breaks jersey-tests-e2e

**Review by breakponchito — APPROVED** [review]
LGTM

**Review by Pandrex247 — COMMENTED** [review]
Running additional tests - I think one of our patches breaks something which passes on 2.36 vanilla.

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
This breaks the osgi-functional tests.
They pass on both 2.36 vanilla and on 2.36 with the cherry-picked OSGi header fix from jansupol.

One of our previous patches breaks this test.

**Comment by kalinchan** [comment]
I managed to successfully run all tests, all of them passed when I ran them.

**Comment by Pandrex247** [comment]
Oh you're right - I done goofed and checked out the wrong branch.

**Review by Pandrex247 — APPROVED** [review]

**Comment by Pandrex247** [comment]
Technically I don't think we actually need the additional changes on top of the microprofile.config.version change in https://github.com/payara/patched-src-jersey/pull/142/commits/c91eff607c334670e2e1bceeeb9473b8af79e41e but that shouldn't block for now.

**Pull request merged**